scripts/provisioning/cleanupcmd.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+import sys
+import shutil
+import ldap
+
+from s3_setup_constants import LDAP_URL
+from s3_setup_constants import LDAP_USER
+
+from s3cipher.cortx_s3_cipher import CortxS3Cipher
+from setupcmd import SetupCmd
+
+class CleanupCmd(SetupCmd):
+  """Cleanup Setup Cmd"""
+  name = "cleanup"
+
+  # map with key as 'account-name', and value as 'constkey-to-generate-access-key' 
+  account_cleanup_dict = {
+                          "s3-background-delete-svc": {
+                            "cipherConstKey": "s3backgroundaccesskey",
+                            "s3userId": "450"
+                          }
+                        }
+
+  config_cleanup_list = ['/opt/seagate/cortx/auth/resources/authserver.properties',
+                      '/opt/seagate/cortx/s3/s3backgrounddelete/config.yaml',
+                      '/opt/seagate/cortx/auth/resources/keystore.properties',
+                      '/opt/seagate/cortx/s3/conf/s3config.yaml']
+  ldap_conn = None
+  
+  def __init__(self, config: str):
+    try:
+      super(CleanupCmd, self).__init__(config)
+      self.read_ldap_credentials()
+
+    except Exception as e:
+      raise e
+
+  def process(self):
+    sys.stdout.write(f"Processing {self.name} {self.url}\n")
+
+    # remove accounts in account_cleanup_list
+    # 1. Check if the account exists or not
+    # 2. Delete the account if it exists, else do nothing
+    self.create_ldap_connection()
+
+    for acc in self.account_cleanup_dict.keys():
+      if(self.if_ldap_account_exists(acc)):
+        sys.stdout.write(f'deleting ldap account: {acc}\n')
+        try:
+          self.delete_ldap_account(acc)
+          self.delete_ldap_connection()
+        except Exception as e:
+          raise e
+      else:
+        sys.stdout.write(f'INFO: ldap account: {acc} does not exist, nothing to do.\n')
+
+    # remove configurations
+    try:
+      sys.stdout.write(f'deleting configuration files\n')
+      self.delete_config_files()
```

Comment:
comment out this call, as this seems to be part of upgrade, which is not clear yet..

Reply:
done